### DISCUSSION TITLE
The url contains '~' seems to be routed to 404

### DISCUSSION THREAD

**Discussion by t3link** [community/community#23991] [Other Feature Feedback, Questions, & Ideas]
<p>Hello, I have a github repo named <code>t3link/t3link.github.io</code>.</p>
<p>Some resource url contains <code>~</code> which the static folder name starts with <code>~</code>.</p>
<p>The request for these resources always return a 404 error.  However it works fine on my local machine.</p>
<p>For example</p>
<ol>
<li>Url1:  <a href="https://t3link.github.io/archives/case-study/%E7%AC%94%E8%AE%B0%E6%9C%AC%E5%8A%A0%E8%A3%85%E5%86%85%E5%AD%98%E9%A3%8E%E6%89%87%E7%AA%81%E7%84%B6%E8%BD%AC%E8%B5%B7%E6%9D%A5%E4%BA%86/" rel="noopener nofollow ugc">https://t3link.github.io/archives/case-study/笔记本加装内存风扇突然转起来了/</a>
</li>
<li>Url2: <a href="https://t3link.github.io/images/~2021-03/hp-zhan-66-4th-r7-5800u.jpg" rel="noopener nofollow ugc">https://t3link.github.io/images/~2021-03/hp-zhan-66-4th-r7-5800u.jpg</a>
</li>
</ol>
<p>Url1 works fine though it contains some Chinese characters, the browser will encode the url automately.<br>
Url2 contains <code>~2021-03</code> and return 404.</p>
<p>What can I do for this problem? Thanks.</p>

**Comment by byrneh** [comment]
<p>HI <a class="mention" href="/t3link">@t3link</a> , welcome to the the community,</p>
<p>On many file systems (Windows for example) , a file name will contain a tilde (~) within component of the name that is too long to comply with 8.3 naming rules.</p>
<p>I generally would advise against using such special character in file names or paths.</p>

**Comment by t3link** [comment, marked as answer]
<p>Thanks, but this may be a last choice :joy:.<br>
Luckily, I found a related topic here by google search.<br>
<a href="https://github.community/t/unable-to-access-resources-in-folder-with-name-starting-with/10505">/t/unable-to-access-resources-in-folder-with-name-starting-with/10505</a>.<br>
According to this solution, I add a empty file named <code>.nojekyll</code> in the folder named <code>static</code> on <code>master</code> branch. Then it will be published to the root directory on <code>gh-pages</code> branch by <a href="https://github.com/getzola/zola" rel="noopener nofollow ugc">zola</a>. Now the url2 with <code>~</code> also works fine.</p>
